packages/react-components/react-motions-preview/src/types.ts

```diff
@@ -5,8 +5,8 @@ export type PresenceMotion = {
   exit: AtomMotion;
 };
 
-export type AtomMotionFn = (element: HTMLElement) => AtomMotion;
-export type PresenceMotionFn = (element: HTMLElement) => PresenceMotion;
+export type AtomMotionFn = (params: { element: HTMLElement }) => AtomMotion;
```

Comment:
I'm not against this change, but I also do not think this is necessary. The animated element is a requirement for animations to work, so that can just be kept as a first argument and future options can be extended in a second parameter

```ts
export type AtomMotionFn = (element: HTMLElement, options: {}) => AtomMotion;
```

Reply:
> The animated element is a requirement for animations to work

In terms of the React wrapper, yes; in terms of the motion atom function, no. A function that generates a **fade** atom can generate opacity keyframes without access to the element. Whereas a motion like **collapse** needs the element to measure its height.

Reply:
In the near future, we will be passing more parameters like `animateOpacity`: many presence motions will auto-fade, and this will optionally turn that off. This is already in the Collapse and Scale implementations in my branch.

Reply:
What is an example of a function generated motion that does not require the element? In cases like opacity I would expect that the static object declaration would also work.

Since these motions are declared outside the lifecycle of react components, I can't really see component state or other app runtime to be used in this function 🤔

Reply:
@ling1726 To be clear, all of these custom parameters will also be custom props for the motion component. They can change at runtime, necessitating a motion function, but access to the element is not involved:

> A preview of what's planned: soon we will be passing more parameters like `animateOpacity`: many presence motions will auto-fade, and this will optionally turn that off. `animateOpacity` is already in the Collapse and Scale implementations in my branch.
> 
> We also want to implement a custom parameter for Scale to control the anchor point. Further down the road, we can see custom params to fade from an opacity other than zero, to collapse to a height other than zero, etc.